framework/src/utils/MooseEnum.C

```diff
@@ -105,6 +105,10 @@ MooseEnum::operator==(const char * name) const
   std::string upper(name);
   std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
 
+  if (std::find(_names.begin(), _names.end(), upper) == _names.end() && (upper != ""))
+    if (_out_of_range_index == 0)     // Are out of range values allowed?
```

Comment:
Well we have to do string comparisons here, but yes I would like to see
this wrapped in the "NDEBUG" constant.

On Fri, Jun 3, 2016 at 9:42 AM Derek Gaston <notifications@github.com>
wrote:

> In framework/src/utils/MooseEnum.C
> <https://github.com/idaholab/moose/pull/7147#discussion_r65728958>:
>
> > @@ -105,6 +105,10 @@ MooseEnum::operator==(const char * name) const
> >    std::string upper(name);
> >    std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
> >
> > +  if (std::find(_names.begin(), _names.end(), upper) == _names.end() && (upper != ""))
> > +    if (_out_of_range_index == 0)     // Are out of range values allowed?
>
> This whole check is incredibly expensive. While you shouldn't be using
> string comparisons in your computational functions (because they are
> already slow)... maybe this new code should only be in debug mode or in an
> assert?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7147/files/431f2255e23f8b45d798d944ae5f254c505ea89b#r65728958>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC5XIJE3pBgjWYm97vnnMYYyXsSlqybUks5qIEtQgaJpZM4ItM09>
> .
>